### PR TITLE
Expanding integration docs

### PR DESCRIPTION
**Pull request by omercnet**
Added a lot of clarity on how to build an integration, we should follow up with a proper guide like we have for services.

Also fixed a bug where alert data was not returned at all if the integration required specific fields to be present.

### CODE REVIEW

File: honeycomb/integrationmanager/integration_utils.py [hunk 1]
```diff
@@ -12,32 +12,64 @@
...
-
-    Will be overridden by output plugins.
-    """
+    """Base Output Integration Class."""
 
     __metaclass__ = ABCMeta
 
     def __init__(self, integration_data):
-        """__init__."""
+        """Use :func:`__init__` to set up any prerequisits needed before sending events, validate paramaters, etc.
```
Comment (Sleeptime17) [inline honeycomb/integrationmanager/integration_utils.py]: prerequisites

File: honeycomb/integrationmanager/integration_utils.py [hunk 2]
```diff
@@ -12,32 +12,64 @@
...
-        """Send event."""
-        pass
+    def send_event(self, alert_dict):
+        """Send alert event to external integration.
+
+        :param alert_dict: A dictionary with all the alert fields.
+        :rtype: tuple(dict(output_data), object(output_file))
+        :raises IntegrationSendEventError: If there's a problem sending the event.
+        :raises IntegrationMissingRequiredFieldError: If a required field is missing.
+        :return: A tuple where the first value is a dictionary with information to display in the UI, and second is and
```
Comment (Sleeptime17) [inline honeycomb/integrationmanager/integration_utils.py]: and [the] second is an[-d] optional file to be atta[c]hed.

**Review by Sleeptime17 — CHANGES_REQUESTED** [review]
I assume the documentation removed is due to a good reason (it's not helpful?)

**Comment by omercnet** [comment]
You mean the validators? The code itself was removed a while ago I forgot to remove the autodocs

**Comment by omercnet** [comment]
@Sleeptime17 please review

**Review by Sleeptime17 — APPROVED** [review]

**Pull request merged**